### DISCUSSION TITLE
Gem5 on Rocky 9

### DISCUSSION THREAD

**Discussion by rgrover1111** [gem5/gem5#918] [Q&A]
Does Gem5 run on Rocky 9?

**Comment by powerjg** [comment]
It's not explicitly tested, but we do support gcc 11. I don't see why we wouldn't support python 3.9, but this is likely also not tested.

Try it, and if it doesn't work, feel free to open an issue with the errors.

**Reply by rgrover1111** [reply]
I followed the following steps to install gem5 on Rocky 9:
1. Run --->git clone https://github.com/gem5/gem5
2. Change directory to gem5 after gem5 software installation --> cd gem5
3. Run  -->  pip install -r requirements.txt
4. Run ---> scons build/{ISA}/gem5.{variant} -j {cpus}    
     example: scons build/X86/gem5.opt -j4

Then I followed the instructions in the following tutorial to write the program hello-world.py:
https://www.gem5.org/documentation/gem5-stdlib/hello-world-tutorial
Upon running the program,  an error is displayed (see attachment).  
![gem_5_type_error](https://github.com/gem5/gem5/assets/162643519/28c95609-3960-4ef6-af50-7d34f9d21e36)

**Reply by powerjg** [reply]
This is not a bug in gem5, but a bug in the documentation. If you look at the code in the upstream repo, there is an extra parameter for the `Processor`.

https://github.com/gem5/gem5/blob/stable/configs/example/gem5_library/arm-hello.py#L65

`processor = SimpleProcessor(cpu_type=CPUTypes.TIMING, isa=ISA.ARM, num_cores=1)`

**Comment by powerjg** [comment]
I'm going to close this discussion as the original question "does gem5 work on rocky 9" has been answered (the answer is "yes" 😄)
